tests/test_types.py

```diff
@@ -1536,7 +1537,10 @@ class Model(BaseModel):
     ],
 )
 def test_invalid_schema_constraints(kwargs, type_):
-    with pytest.raises(SchemaError, match='Invalid Schema:\n.*\n  Extra inputs are not permitted'):
+    match = (
+        r'(:?Invalid Schema:\n.*\n  Extra inputs are not permitted)|(:?The following constraints cannot be applied to)'
+    )
+    with pytest.raises((SchemaError, TypeError), match=match):
```

Comment:
It's now a mix, depending on where the error is coming from. It will be just the latter since my long term plan is for constraints to not automatically insert themselves into the core schema blindly, instead they will check if the core schema they see accepts them as constraints and if not insert a Python function validator version of the consteraint.